NBitcoin/Transaction.cs

```diff
@@ -1547,11 +1547,8 @@ public virtual void ReadWrite(BitcoinStream stream)
 				/* Try to read the vin. In case the dummy is there, this will be read as an empty vector. */
 				stream.ReadWrite(ref vin);
 				vin.Transaction = this;
-				var hasNoDummy = (nVersion & NoDummyInput) != 0 && vin.Count == 0;
-				if (witSupported && hasNoDummy)
-					nVersion = nVersion & ~NoDummyInput;
 
-				if (vin.Count == 0 && witSupported && !hasNoDummy)
+				if (vin.Count == 0 && witSupported)

```

Comment:
Not super familiar with the codebase, but the version based check is present below when writing out a transaction. As a result, I think if you tried to write out a segwit transaction with that version that tripped things up in the first place, it may end up writing it out as a non-segwit txn.

Reply:
Either way, I'd recommend treating the flag market bytes and the version as distinct components during parsing.

Reply:
This whole dummy things shouldn't have been in the source anymore... I'm now copying the Bitcoin Core implementation to make sure it match.

But this broke my tests, due to parsing for transactions without input... Normally this can't happen in the network, but internally in NBitcoin it happens time to time. So I'm just adapting for this case only.